controllers/ProposalController.php

```diff
@@ -418,4 +439,126 @@ private function saveProposalRelatedFile(string $movedFilename, Proposal $propos
+     */
+    public function actionEditProposal()
+    {
+        $editedProposal = Proposal::findOne(['id' => Yii::$app->request->get()]);
+        $lastProposalContent = $editedProposal->proposalContentHistories[
+        count($editedProposal->proposalContentHistories)-1];
+        $model = New ManageProposalForm();
+        $postRequest = Yii::$app->request->post();
+
+        $model->title = $postRequest['ManageProposalForm']['title'];
+        $model->content = $postRequest['ManageProposalForm']['content'];
+
+        if ($model->load(Yii::$app->request->post()) && $model->validate()) {
+            $transaction = Yii::$app->db->beginTransaction();
+            try {
+                $this->saveEditedProposal($postRequest['ManageProposalForm']['title'], $editedProposal);
+
+                if ($postRequest['ManageProposalForm']['content'] != $lastProposalContent->content) {
+                    $this->saveProposalContent($postRequest['ManageProposalForm']['content'], $editedProposal, false);
+                }
+
+                if (!empty($_FILES['ManageProposalForm']['name']['relatedFile'])) {
+
+                    if (!is_null($editedProposal->files[0]->path)) {
+                        $this->removeExistingUploadedFile($editedProposal->files[0]->path);
+                    }
+
+                    $uploadedFile = $_FILES['ManageProposalForm'];
+                    $movedFilename = $this->moveUploadedFileToServer($uploadedFile, $editedProposal);
+                    $this->saveEditedFile($movedFilename, $editedProposal);
+                    $this->saveProposalFileHistory($movedFilename, $editedProposal, false);
+                }
+
+                $transaction->commit();
+                return $this->redirect('/proposal/my-proposals/' . $editedProposal->id);
+            } catch (CannotHandleUploadedFileException $cannotHandleUploadedFileException) {
+                $transaction->rollBack();
+                return $this->redirect('/proposal/my-proposals/' . $editedProposal->id);
+
+            } catch(\Throwable $e) {
+                $transaction->rollBack();
+                throw $e;
+            }
+        } else {
+            return $this->redirect('/proposal/my-proposals/' . $editedProposal->id);
+        }
+    }
+
+    /**
+     * Save the edited Proposal in DB.
+     *
+     * @param string $proposalTitle
+     * @param Proposal $proposal
+     * @throws CannotSaveException
+     */
+    private function saveEditedProposal(string $proposalTitle, Proposal $proposal)
+    {
+        $proposal->title = $proposalTitle;
+
+        if(!$proposal->save()) {
+            throw new CannotSaveException($proposal);
+        }
+    }
+
+    /**
+     * Save the edited file in DB
+     *
+     * @param $movedFilename
+     * @param Proposal $proposal
+     * @throws CannotSaveException
+     */
+    private function saveEditedFile($movedFilename, Proposal $proposal)
+    {
+        $file = $proposal->files[0];
```

Comment:
We have to edit the database model : a proposal have maximum 1 file, so the field proposal_id in the file table must be unique

Reply:
After that, you will have to use $file = $proposal->file